### PR TITLE
Upgrade to Apollo v3

### PR DESCRIPTION
**Pull request by vinnyjth**
**IMPORTANT: If this is for testing code in our npm dependencies, you should be pointing your PR at the `canary` or `next` branch. `master` should never rely on unreleased code.**

### CODE REVIEW

File: apollos-church-api/package.json [hunk 1]
```diff
@@ -16,7 +16,7 @@
     "lint": "eslint ./src --ext .js"
   },
   "engines": {
-    "node": ">=10.19.x"
+    "node": ">=8.x.x"
```
Comment (vinnyjth) [inline apollos-church-api/package.json:19]: This can/should probably be backed out

Comment (redreceipt) [reply apollos-church-api/package.json:19]: so you want me to undo that change? I'm confused

File: apolloschurchapp/index.js [hunk 2]
```diff
@@ -1,8 +1,12 @@
 import './loadConfig';
 import { AppRegistry, YellowBox } from 'react-native';
 import ApollosConfig from '@apollosproject/config';
-import Storybook from './storybook';
 
+// temp fix for the promise.finally
+// https://github.com/storybookjs/storybook/issues/8371
+const fn = Promise.prototype.finally;
+const Storybook = require('./storybook').default; // eslint-disable-line
+Promise.prototype.finally = fn; // eslint-disable-line
```
Comment (redreceipt) [inline apolloschurchapp/index.js:9]: what was this about?

File: apolloschurchapp/package.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
   "version": "1.0.0",
   "private": true,
   "engines": {
-    "node": ">=8.11.x"
+    "node": ">=8.x.x"
```
Comment (redreceipt) [inline apolloschurchapp/package.json:6]: is there a reason we shouldn't bump to 10? it's LTS and this is a 2.0 branch

File: apolloschurchapp/schema.graphql [hunk 4]
```diff
@@ -1,742 +0,0 @@
-directive @cacheControl(maxAge: Int, scope: CacheControlScope) on FIELD_DEFINITION | OBJECT | INTERFACE
```
Comment (redreceipt) [inline apolloschurchapp/schema.graphql:1]: do we not need this file in `src/client/__mocks__` anymore?

Comment (redreceipt) [reply apolloschurchapp/schema.graphql:1]: or did you just `git rm` it so it's not in the repo anymore?

**Comment by redreceipt** [comment]
@vinnyjth works. I asked a couple of low-importance questions above, I'll work on fixing tests, and then I'll approve. I'm excited to play around with the other improvements that AC3 brings

File: apolloschurchapp/index.js (hunk 2 above)
Comment (vinnyjth) [reply apolloschurchapp/index.js:9]: see comment above: 

https://github.com/storybookjs/storybook/issues/8371

**Review by redreceipt — APPROVED** [review]

**Pull request merged**